### PR TITLE
fix(gradle): Parse dependencies declared via `library()`

### LINKED ISSUE #15398: Rennovate does not recognize dependencies declared in gradle version catalog
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Rennovate fails to resolve all of my dependencies because it thinks my declared alias is the group and my declared group is the name.

Example:

`Failed to look up dependency spring.boot.configuration.processor:org.springframework.boot`

```
version("spring.boot", "2.5.11")
library("spring.boot.configuration.processor", "org.springframework.boot", "spring-boot-configuration-processor").versionRef("spring.boot")
```

### Relevant debug logs

<details><summary>Logs</summary>

```
{
            "depName": "spring.boot.configuration.processor:org.springframework.boot",
            "currentValue": "spring-boot-configuration-processor",
            "managerData": {
              "fileReplacePosition": 1858,
              "packageFile": "settings.gradle"
            },
            "fileReplacePosition": 1858,
            "registryUrls": [
              "https://repo.maven.apache.org/maven2"
            ],
            "depIndex": 1,
            "updates": [],
            "warnings": [
              {
                "topic": "spring.boot.configuration.processor:org.springframework.boot",
                "message": "Failed to look up dependency spring.boot.configuration.processor:org.springframework.boot"
              }
            ],
            "versioning": "gradle"
          },
...
"https://repo.maven.apache.org/maven2/spring/boot/configuration/processor/org.springframework.boot/index.html (GET,404)": 1,
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

New syntax variation:
```gradle
version("spring.boot", "2.5.11")
library("spring.boot.configuration.processor", "org.springframework.boot", "spring-boot-configuration-processor").versionRef("spring.boot")
```

## Context

- Closes: #15398

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
https://github.com/renovate-testing/test-15398-gradle-library/pull/1/files

File: lib/modules/manager/gradle/parser.ts [hunk 1]
```diff
@@ -349,6 +349,34 @@ function processLongFormDep({
...
+  const fileReplacePosition = varNameToken.offset;
+  const packageFile = input.packageFile;
+
+  const groupId = tokenMap.groupId?.value;
+  const artifactId = tokenMap.artifactId?.value;
+  const value = `${groupId}:${artifactId}`;
+  const res: SyntaxHandlerOutput =
+    groupId && artifactId
+      ? { vars: { [key]: { key, value, fileReplacePosition, packageFile } } }
+      : {};
```
Comment (viceice) [inline lib/modules/manager/gradle/parser.ts:366]: missing coverage

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/gradle/parser.ts (hunk 1 above)
Comment (viceice) [reply lib/modules/manager/gradle/parser.ts:366]: probably needs ignore comment as it's never happening?

**Review by rarkins — APPROVED** [review]

**Pull request merged**